tests/gacha_tests.cpp

```diff
@@ -0,0 +1,270 @@
+#include "account_tester.hpp"
+#include "util.hpp"
+
+struct extended_name {
+   name name;
+   struct name contract;
+};
+
+FC_REFLECT(extended_name, (name)(contract))
+
+struct grade {
+   asset                    reward;
+   uint32_t                 score;
+   fc::optional<uint32_t>   limit;
+};
+
+namespace fc {
+   inline void to_variant(const grade& var, fc::variant& vo) {
+      vo = fc::mutable_variant_object()("reward", var.reward)("score", var.score)("limit", var.limit ? *var.limit : fc::variant());
+   }
+}
+FC_REFLECT(grade, (reward)(score)(limit))
+
+std::ostream& operator<<( std::ostream& osm, const fc::mutable_variant_object& v ) {
```

Comment:
I added this because mutable_variant_object can be casted to variant_object implcitly, but not true. Explicit type cast to variant_object is used in following code, so this will not be necessary.